libp2p/protocols/mix/exit_layer.nim

```diff
@@ -0,0 +1,29 @@
+import chronicles, chronos, metrics
+import ../../builders
+import ../../stream/connection
+import ./mix_metrics
+
+type ExitLayer* = object
+  switch: Switch
+
+proc init*(T: typedesc[ExitLayer], switch: Switch): T =
+  ExitLayer(switch: switch)
+
+proc onMessage*(
+    self: ExitLayer,
+    codec: string,
+    message: seq[byte],
+    destAddr: MultiAddress,
+    destPeerId: PeerId,
+) {.async: (raises: [CancelledError]).} =
+  var destConn: Connection
+  try:
+    destConn = await self.switch.dial(destPeerId, @[destAddr], codec)
+    await destConn.write(message)
+  except CatchableError as e:
```

Comment:
missing CancelledError propagation
also it's always better to list specific errors, in case interface (raised errors) changes.

Reply:
Added the CancelledError propagation. For the other exceptions, added a comment in code indicating why they're not being raised